alibi_detect/utils/fetching.py

```diff
@@ -404,14 +406,36 @@ def fetch_state_dict(url: str, filepath: Union[str, os.PathLike],
     -------
     Detector metadata and state.
     """
-    # fetch and save metadata and state dict
-    # TODO - Currently load pickle from bucket and save locally as dill. Convert to dill on bucket?
-    filepath = Path(filepath)
-    path_meta = os.path.join(url, 'meta.pickle')
-    meta = dill.load(urlopen(path_meta))
-    path_state = os.path.join(url, meta['name'] + '.pickle')
-    state_dict = dill.load(urlopen(path_state))
+    # Check if metadata stored as dill or pickle
+    try:
+        url_meta = os.path.join(url, 'meta.dill')
+        resp = requests.get(url_meta, timeout=2)
+        resp.raise_for_status()
+        suffix = '.dill'
+    except RequestException:
+        try:
+            url_meta = os.path.join(url, 'meta.pickle')
+            resp = requests.get(url_meta, timeout=2)
+            resp.raise_for_status()
+            suffix = '.pickle'
+        except RequestException:
+            logger.exception('Timed out which searching for meta.dill or meta.pickle files at {}.'.format(url))
```

Comment:
which --> while

I was going to suggest using f-strings, but then I remembered (found?) this: https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/ Basically it suggests using %-formatting with passing values as arguments to the `exception` method over f-strings or str.format.